digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala

```diff
@@ -224,68 +289,110 @@ trait LaneOperations {
                     else
                       Seq()
 
+        val (laneCodeStart, laneCodeEnd) = getLaneNumbersRange(mainLaneDirectionOK)
+        val needAdjustSideCode = getLaneCodesForSideCodeAdjust(lanesToProcess, laneCodeStart, laneCodeEnd, SideCode.BothDirections)
+
         val toRemove = lanesToProcess.filterNot(_.id == 0L)
                                     .filter(lane => (lane.laneCode >= mainLaneDirectionRemove._1 && lane.laneCode <= mainLaneDirectionRemove._2) ||
                                               lane.sideCode != SideCode.BothDirections.value)
                                     .map(_.id)
 
 
         val lanesToAdd = (lanes ++ toAdd).filterNot(lane => toRemove.contains(lane.id))
+        val newChangeSet = changeSet.copy(generatedPersistedLanes = changeSet.generatedPersistedLanes ++ lanesToAdd,
+                                    expiredLaneIds = changeSet.expiredLaneIds ++ toRemove,
+                                    adjustedSideCodes = changeSet.adjustedSideCodes ++ needAdjustSideCode )
 
-        (lanesToAdd, changeSet.copy(expiredLaneIds = changeSet.expiredLaneIds ++ toRemove, generatedPersistedLanes = changeSet.generatedPersistedLanes ++ lanesToAdd))
+        (lanesToAdd, newChangeSet)
 
 
       case TrafficDirection.AgainstDigitizing =>
 
-        val toAdd = if (!lanesToProcess.exists(lane => lane.laneCode == mainLaneDirectionOK && lane.sideCode == SideCode.BothDirections.value))
+        val toAdd = if (!lanesToProcess.exists(lane => lane.laneCode == mainLaneDirectionOK))
                       Seq(createPersistedLane(mainLaneDirectionOK, SideCode.BothDirections.value, baseLane.municipalityCode, baseProps))
                     else
                       Seq()
 
+        val (laneCodeStart, laneCodeEnd) = getLaneNumbersRange(mainLaneDirectionOK)
+        val needAdjustSideCode = getLaneCodesForSideCodeAdjust(lanesToProcess, laneCodeStart, laneCodeEnd, SideCode.BothDirections)
+
         val toRemove = lanesToProcess.filterNot(_.id == 0L)
                                     .filter(lane => (lane.laneCode >= mainLaneDirectionRemove._1 && lane.laneCode <= mainLaneDirectionRemove._2) ||
                                               lane.sideCode != SideCode.BothDirections.value)
                                     .map(_.id)
 
 
         val lanesToAdd = (lanes ++ toAdd).filterNot(lane => toRemove.contains(lane.id))
+        val newChangeSet = changeSet.copy(generatedPersistedLanes = changeSet.generatedPersistedLanes ++ lanesToAdd,
+                                    expiredLaneIds = changeSet.expiredLaneIds ++ toRemove,
+                                    adjustedSideCodes = changeSet.adjustedSideCodes ++ needAdjustSideCode )
 
-        (lanesToAdd, changeSet.copy(expiredLaneIds = changeSet.expiredLaneIds ++ toRemove, generatedPersistedLanes = changeSet.generatedPersistedLanes ++ lanesToAdd))
+        (lanesToAdd, newChangeSet)
```

Comment:
this case is equal to last one, just do it like this 
      case TrafficDirection.TowardsDigitizing | TrafficDirection.AgainstDigitizing =>

Reply:
Done